chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -254,8 +254,12 @@ def self.all(node)
           node[:block_device].keys.map{|d|Disk.new(node,d)}
         end
 
-        def self.unclaimed(node)
+        def self.unclaimed(node,mounted=false)
           all(node).select do |d|
+            unless mounted
+              %x{lsblk #{d.name} --noheadings --output MOUNTPOINT | grep -q -v ^$}
```

Comment:
grep should exit non-0 if there are only empty lines.

The code is stolen from bootdisk.rb